### PR TITLE
Better Serializing

### PR DESCRIPTION
**Pull request by blainsmith**
There was issues when relying on the object map when saving entities so I used a different serializer before saving to remove the need for storing an object map.

### CODE REVIEW

**Comment by blainsmith** [comment]
The tests pass, but coverage isn't happy.

**Comment by mihaidma** [comment]
@blainsmith what issues did you encounter? what problem did you try to solve?

**Comment by blainsmith** [comment]
There were cases when records didn't match the object map and they were not able to be fetch again. This removes the the global object map for entities.

**Comment by genexp** [comment]
This is huge improvement.  The current redis-store plugin locks schema for any entity the very first time it sees that entity.  This works differently than say, the mongo store, which allows you to store any arbitrary schema for a given entity.  Doing this is an awful hack, and seemed to be in place only to ensure that dates get serialized in a reversible way.  This patch fixes that.

**Comment by mihaidma** [comment]
I'm currently removing the camel case to pascal case conversion from mysql and postgres, this week will release the new versions. After that I will look also at redis. Your change looks good, I will check if it fits with the general store approach.

File: redis-store.js [hunk 1]
```diff
@@ -160,17 +156,13 @@ module.exports = function (opts) {
         }
       }
 
-      entp = makeentp(ent)
+      entp = Cryo.stringify(ent.data$(false))
 
-      var objectMap = determineObjectMap(ent)
+      // var objectMap = determineObjectMap(ent)
```
Comment (mihaidma) [inline redis-store.js]: please remove comment

File: redis-store.js [hunk 2]
```diff
@@ -160,17 +156,13 @@ module.exports = function (opts) {
         }
       }
 
-      entp = makeentp(ent)
+      entp = Cryo.stringify(ent.data$(false))
```
Comment (mihaidma) [inline redis-store.js]: :+1:

**Comment by mihaidma** [comment]
@blainsmith 
I verified cryo, seems good
looked also over the code and made some tests, seems ok to me

could you please rebase?

**Comment by blainsmith** [comment]
I forgot this was out there actually if you want there is a smaller serializer lib that I wrote too that just wraps the native `JSON.stringify` and `JSON.parse` so you don't have a dependency to read the data in Redis from other sources.

I could swap it out quick if you want? https://www.npmjs.com/package/nosj

**Comment by mihaidma** [comment]
nosj looks good, please swap.

**Comment by blainsmith** [comment]
I think the test suite is really broken now. For some reason that I haven't had time to investigate it the `seneca-store-test` dependency is not working properly.

**Comment by mihaidma** [comment]
i tested it yesterday, it is working. Please use the 1.0.0 test suite version

**Comment by mihaidma** [comment]
I looked over your latest code changes, seem ok
I can take over to finalize just please solve the conflicts so that I can merge

**Comment by mihaidma** [comment]
LGTM, thank you

**Comment by mihaidma** [comment]
awesome work :+1:

**Pull request merged**